presto-main/src/main/java/com/facebook/presto/execution/SqlTask.java

```diff
@@ -373,6 +375,11 @@ private TaskInfo createTaskInfo(TaskHolder taskHolder)
         return Futures.transform(futureTaskState, input -> getTaskInfo(), directExecutor());
     }
 
+    public void updateCommunicationSlug(String communicationSlug)
+    {
+        this.communicationSlug.compareAndSet(null, requireNonNull(communicationSlug, "communicationSlug is null"));
```

Comment:
```
boolean isSuccessfullySet = this.communicationSlug.compareAndSet(null, requireNonNull(communicationSlug, "communicationSlug is null"));
verify(isSuccessfullySet, "communicationSlug already set");
```

Reply:
I see why you need to do update instead of set and silently ignore the new updates. I would prefer there is a way we can actually validate this value haven only been set at task creation. 

If there is not a good way, then probably add a comment here and also add test to make sure the value of communicationSlug will not change once set (so that later people making the change knows the behavior).